flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/sink/SinkRecord.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.store.connector.sink;
+
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.data.binary.BinaryRowData;
+import org.apache.flink.types.RowKind;
+
+import static org.apache.flink.util.Preconditions.checkArgument;
+
+/** A sink records contains key, value and partition, bucket, row kind information. */
+public class SinkRecord {
+
+    private final BinaryRowData partition;
+
+    private final int bucket;
+
+    private final RowKind rowKind;
+
+    private final BinaryRowData key;
+
+    private final RowData row;
+
+    public SinkRecord(
+            BinaryRowData partition, int bucket, RowKind rowKind, BinaryRowData key, RowData row) {
+        checkArgument(partition.getRowKind() == RowKind.INSERT);
+        checkArgument(key.getRowKind() == RowKind.INSERT);
+        checkArgument(row.getRowKind() == RowKind.INSERT);
```

Comment:
Why do we need this check? We have no concern on the row kind of each `BinaryRowData`. We only need the `rowKind` given in the argument.

Reply:
We should avoid the caller to determine by this kind, the caller should use member variables.
LogStore also use this class.